internal/transformer/classes/generators/dockerfilegenerators/winconsoleappdockerfilegenerator.go

```diff
@@ -0,0 +1,185 @@
+	if err != nil {
+		return nil
+	}
+
+	defer appConfigFile.Close()
+
+	byteValue, _ := ioutil.ReadAll(appConfigFile)
+	appCfg := dotnet.AppConfig{}
+	xml.Unmarshal(byteValue, &appCfg)
+	if err != nil {
+		logrus.Errorf("Could not parse the App.config file: %s", err)
+		return nil
+	}
+
+	ports := make([]string, 0)
+	for _, addKey := range appCfg.AppCfgSettings.AddList {
+		if parsedUrl, err := url.ParseRequestURI(addKey.Value); err == nil && parsedUrl.Scheme != "" && parsedUrl.Host != "" {
+			_, port, _ := net.SplitHostPort(parsedUrl.Host)
+			if err == nil {
+				ports = append(ports, port)
+			}
+		}
+	}
+
+	if len(appCfg.Model.Services.ServiceList) > 0 {
+		for _, svc := range appCfg.Model.Services.ServiceList {
+			for _, addKey := range svc.Host.BaseAddresses.AddList {
+				if parsedUrl, err := url.ParseRequestURI(addKey.BaseAddress); err == nil && parsedUrl.Scheme != "" && parsedUrl.Host != "" {
+					_, port, _ := net.SplitHostPort(parsedUrl.Host)
+					if err == nil {
+						ports = append(ports, port)
+					}
+				}
+			}
+		}
+	}
+
+	return ports
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *WinConsoleAppDockerfileGenerator) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	dirEntries, err := os.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Error while trying to read directory: %s", err)
+		return nil, nil, err
+	}
+	appName := ""
+	ports := make([]string, 0)
+	for _, de := range dirEntries {
+		ext := filepath.Ext(de.Name())
+		if ext != dotnet.CsSln {
+			continue
+		}
+		csProjPaths := parseSolutionFile(filepath.Join(dir, de.Name()))
+
+		if csProjPaths == nil || len(csProjPaths) == 0 {
+			logrus.Errorf("No projects available for the solution: %s", de.Name())
+			continue
+		}
+
+		for _, csPath := range csProjPaths {
+			projPath := filepath.Join(strings.TrimSpace(dir), strings.TrimSpace(csPath))
+			byteValue, err := ioutil.ReadFile(projPath)
+			if err != nil {
+				logrus.Errorf("Could not read the project file: %s", err)
+				continue
+			}
+
+			configuration := dotnet.CSProj{}
+			err = xml.Unmarshal(byteValue, &configuration)
+			if err != nil {
+				logrus.Errorf("Could not parse the project file: %s", err)
+				continue
+			}
+
+			if configuration.PropertyGroup == nil ||
+				configuration.PropertyGroup.TargetFrameworkVersion == "" ||
+				!dotnet.FourXPattern.MatchString(configuration.PropertyGroup.TargetFrameworkVersion) {
+				logrus.Errorf("Not a supported dotnet framework [%s]", configuration.PropertyGroup.TargetFrameworkVersion)
+				continue
+			}
+
+			isWebProj, err := isWeb(configuration)
+			if err != nil || isWebProj {
+				continue
+			}
+
+			portList := t.parseAppConfig(filepath.Join(dir, filepath.Dir(csPath)))
+			if portList != nil {
+				ports = append(ports, portList...)
+			}
+
+			appName = strings.TrimSuffix(filepath.Base(de.Name()), filepath.Ext(de.Name()))
```

Comment:
If there is more than one solution file, the behaviour is undeterministic now. We should think about how we should act in that case

Reply:
@ashokponkumar More than one solution file for project is highly unlikely.

Reply:
Sure, In that case we should break after processing a single solution file.

Reply:
done